### DISCUSSION TITLE
Database partially migrated

### DISCUSSION THREAD

**Discussion by Ronnnn** [postalserver/postal#3039] [Help with using Postal] [answered]
## Describe the bug

I've migrated from an 2019 version v1 to the newest 3.x using the migration guide. The Database however is not fully migrated as some rows are missing. For example:

```
describe webhook_requests;
+-------------+--------------+------+-----+---------+----------------+
| Field       | Type         | Null | Key | Default | Extra          |
+-------------+--------------+------+-----+---------+----------------+
| id          | int(11)      | NO   | PRI | NULL    | auto_increment |
| server_id   | int(11)      | YES  |     | NULL    |                |
| webhook_id  | int(11)      | YES  |     | NULL    |                |
| url         | varchar(255) | YES  |     | NULL    |                |
| event       | varchar(255) | YES  |     | NULL    |                |
| uuid        | varchar(255) | YES  |     | NULL    |                |
| payload     | text         | YES  |     | NULL    |                |
| attempts    | int(11)      | YES  |     | 0       |                |
| retry_after | datetime(6)  | YES  |     | NULL    |                |
| error       | text         | YES  |     | NULL    |                |
| created_at  | datetime(6)  | YES  |     | NULL    |                |
+-------------+--------------+------+-----+---------+----------------+
```

Which should be:

```
create_table "webhook_requests", id: :integer, charset: "utf8mb4", collation: "utf8mb4_general_ci", force: :cascade do |t|
    t.integer "server_id"
    t.integer "webhook_id"
    t.string "url"
    t.string "event"
    t.string "uuid"
    t.text "payload"
    t.integer "attempts", default: 0
    t.datetime "retry_after"
    t.text "error"
    t.datetime "created_at"
    t.string "locked_by"
    t.datetime "locked_at", precision: nil
    t.index ["locked_by"], name: "index_webhook_requests_on_locked_by"
  end

```
## Question

What would be the best way to fix this on the live server? I runned postal upgrade once before. But the binlog_format was set to STATEMENT at that point. I did update the ROW_FORMAT in all postal tables though to dynamic. Should I just run postal upgrade again?

**Comment by willpower232** [comment, marked as answer]
Those fields are added by a migration so its reasonable to assume that they haven't been run properly

https://github.com/postalserver/postal/blob/main/db/migrate/20240214132253_add_lock_fields_to_webhook_requests.rb

Looking at the helper script https://github.com/postalserver/install/blob/main/bin/postal it looks like you can run `postal upgrade` repeatedly however you may also choose to just run `postal upgrade-db` to skip out the bit that messes with docker.

**Reply by Ronnnn** [reply]
Thank you for this. I runned the upgrade-db and this gave me a hint on what is going on:

```
Caused by:
Mysql2::Error: Specified key was too long; max key length is 767 bytes (Mysql2::Error)
```

So I decided to go along with the dist-update on the live server. The release update wasn't possible to begin with, because of the broken rabbitmq dependency. The release update will bump the mariadb version from 10.1 to 10.3 which will allow for larger key lengths. And yes this fixed the problem, and `postal upgrade-db` runned without errors. 

So anyone in the same situation:
- stop postal
- remove rabbitmq
- do the dis-update
- update postal
